cosmos/dbt/selector.py

```diff
@@ -295,7 +295,14 @@ def _should_include_node(self, node_id: str, node: DbtNode) -> bool:
         self.visited_nodes.add(node_id)
 
         if node.resource_type == DbtResourceType.TEST:
-            node.tags = getattr(self.nodes.get(node.depends_on[0]), "tags", [])
+            dependency_ids = [node_id for node_id in node.depends_on if node_id.startswith("model.")]
+            parent_id = dependency_ids[-1]
+            if len(dependency_ids) > 1:
+                logger.warning(
+                    f"Test node {node.name} has more than one model dependency {dependency_ids}, selected tags from parent assumed to be {parent_id}."
```

Comment:
I liked that you added this warning. Before users were left out about what was happening.